File: osmdroid-android/src/main/java/org/osmdroid/util/PointL.java
package org.osmdroid.util;

/**
 * A {@link android.graphics.Point} with coordinates in long type instead of int
 */

public class PointL {

    public long x;
    public long y;

    public PointL() {
    }

    public PointL(final long pX, final long pY) {
        x = pX;
        y = pY;
    }

    public PointL(final PointL pOther) {
        set(pOther);
    }

    public void set(final PointL pOther) {
        x = pOther.x;
        y = pOther.y;
    }

    public void set(final long pX, final long pY) {
        x = pX;
        y = pY;
    }

    public final void offset(long dx, long dy) {
        x += dx;
        y += dy;
    }

    @Override
    public String toString() {
        return "PointL(" + x + ", " + y + ")";
    }

    @Override
    public boolean equals(Object object) {
        if (this == object) {
            return true;
        }
        if (!(object instanceof PointL)) {
            return false;
        }
        final PointL other = (PointL) object;
        return x == other.x && y == other.y;
    }
}

File: osmdroid-android/src/main/java/org/osmdroid/util/PointL.kt
package org.osmdroid.util

/**
 * A [android.graphics.Point] with coordinates in long type instead of int
 */
class PointL {
    @JvmField
    var x: Long = 0

    @JvmField
    var y: Long = 0

    constructor()

    constructor(pX: Long, pY: Long) {
        x = pX
        y = pY
    }

    constructor(pOther: PointL) {
        set(pOther)
    }

    fun set(pOther: PointL) {
        x = pOther.x
        y = pOther.y
    }

    fun set(pX: Long, pY: Long) {
        x = pX
        y = pY
    }

    fun offset(dx: Long, dy: Long) {
        x += dx
        y += dy
    }

    override fun toString(): String {
        return "PointL(" + x + ", " + y + ")"
    }

    override fun equals(`object`: Any?): Boolean {
        if (this === `object`) {
            return true
        }
        if (`object` !is PointL) {
            return false
        }
        val other = `object`
        return x == other.x && y == other.y
    }
}
